src/providers/dataProvider/index.ts

```diff
@@ -78,33 +78,76 @@ const withCreatedBy = (
 }
 
 const customMethods = (provider: DataProvider): CustomDataProvider => {
+  const audit = trackEvent(provider)
+  const user = getUser()
+  const { name: userName = '' } = user ?? { name: '' }
+
   return {
-    loanItems: async (
-      items: Array<Item['id']>,
-      recipient: User['id'],
-      loan: Loan['id']
-    ) => {
-      const promisees = items.map(async (item) => {
-        const data: Partial<LoanItem> = {
-          receivedBy: recipient,
-          loan,
-          item,
-          createdAt: nowDate()
+    loanItems: async (items: Array<Item['id']>, loan: Loan) => {
+      const { holder, remarks = '' } = loan
+
+      await provider.updateMany<Item>(constants.R_ITEMS, {
+        ids: items,
+        data: {
+          loanedBy: holder
         }
-        await provider.create<LoanItem>(constants.R_LOAN_ITEMS, {
-          data
+      })
+
+      const {
+        data: { name }
+      } = await provider.getOne<User>(constants.R_USERS, {
+        id: holder
+      })
+
+      const promisees = items.map(async (item) => {
+        await audit({
+          type: AuditType.ITEM_LOAN,
+          activityDetail: `Item loaned to ${name} by ${userName}. ${remarks}`,
+          resource: constants.R_LOAN_ITEMS,
```

Comment:
This should be `constants.R_ITEMS`

We should probably delete `constants.R_LOAN_ITEMS`